controllers/clustertopology_controller_compute.go

```diff
@@ -308,6 +332,101 @@ func computeCluster(current *clusterTopologyState, infrastructureCluster, contro
 	return cluster
 }
 
+// computeMachineDeploymentTopologyState computes the desired state for a MachineDeploymentTopology
+func computeMachineDeploymentTopologyState(class *clusterTopologyClass, current *clusterTopologyState, machineDeploymentTopology clusterv1.MachineDeploymentTopology, currentMachineDeployment *clusterv1.MachineDeployment) (machineDeploymentTopologyState, error) {
+	state := machineDeploymentTopologyState{}
+
+	className := machineDeploymentTopology.Class
+	machineDeploymentClass := getMachineDeploymentClass(class, className)
+
+	machineDeploymentTopologyClass, ok := class.machineDeployments[className]
+	if !ok {
+		return state, errors.Errorf("class name %s not found", className)
+	}
+
+	state.bootstrapTemplate = templateToTemplate(templateToInput{
+		template:              machineDeploymentTopologyClass.bootstrapTemplate,
+		templateClonedFromRef: objToRef(machineDeploymentTopologyClass.bootstrapTemplate),
+		cluster:               current.cluster,
+		namePrefix:            fmt.Sprintf("%s-%s-bootstrap-", current.cluster.Name, machineDeploymentTopology.Name),
+		currentObjectRef:      machineDeploymentClass.Template.Bootstrap.Ref,
```

Comment:
@fabriziopandini I could be wrong, but I think we need to get the currentObjectRef somewhere from current. and not from the refs in the ClusterClass

(same would apply for InfrastructureMachineTemplate below)

Reply:
good catch, you are right!

Reply:
Nice catch, thanks!